src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import duke.exception.DeadlineCommandUseException;
+import duke.exception.EventCommandUseException;
+import duke.exception.InvalidInputException;
+import duke.exception.ToDoCommandUseException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.Serializable;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+public class Parser implements Serializable {
+
+    public Parser() {
+        //empty constructor to initialize class objects
+    }
+
+    public void chat(String str, TaskList tasks) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+        Ui ui = new Ui();
+        try {
+                if (!str.equals("list")) {
+                    if (str.startsWith("mark ")) {
+                        String num = str.substring(5);
+                        int number = Integer.valueOf(num);
+                        if (number <= 0 || number > tasks.getSize()) {
+                            throw new InvalidInputException(str);
+                        }
+                        int index = number - 1; //index for task list
+                        tasks.markDone(index);
+                        Task done = tasks.getTask(index);
+                        ui.printDone(done);
+                    } else if (str.startsWith("unmark ")) {
+                        String num = str.substring(7);
+                        int number = Integer.valueOf(num);
+                        if (number <= 0 || number > tasks.getSize()) {
+                            throw new InvalidInputException(str);
+                        }
+                        int index = number - 1; //index for task list
+                        tasks.markNotDone(index);
+                        Task notDone = tasks.getTask(index);
+                        ui.printNotDone(notDone);
+                    } else if (str.startsWith("delete ")) {
+                        String num = str.substring(7);
+                        int number = Integer.valueOf(num);
+                        if (number <= 0 || number > tasks.getSize()) {
+                            throw new InvalidInputException(str);
+                        }
+                        int index = number - 1;
+                        Task toBeDeleted = tasks.removeTask(index);
+                        ui.printDelete(toBeDeleted, tasks);
+                    } else {
+                        if (str.startsWith("todo")) {
+                            String todo = str.substring(4);
+                            //remove any leading and trailing whitespace characters and
```

Comment:
Great on abiding the rule that comments should be indented relative to their position in the code! 😊